lib/mix/ecto.ex

```diff
@@ -104,6 +104,87 @@ defmodule Mix.Ecto do
   end
 
   @doc """
+  Gets the storage path from the repository.
+  """
+  @spec storage_path(Ecto.Repo.t) :: String.t
+  def storage_path(repo) do
+    repo_priv(repo)
+  end
+
+  @doc """
+  Add me
+  """
+  @spec storage_filename(Ecto.Repo.t) :: String.t
+  def storage_filename(repo) do
+    database_info = repo.__adapter__.database_info
+    "#{inspect(repo)}-#{database_info.type}-#{database_info.version}.storage"
```

Comment:
I think the whole storage_filename should either be part of the adapter API (instead of `database_info`) or it should be simply transparent (i.e. the adapter only imports schema_load/schema_dump and is responsible for checking if the file exists). I am more inclined towards the second but if you prefer some other approach, then please let me know. :)